daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/RunnerFactory.scala

```diff
@@ -91,75 +98,127 @@ object Runner {
  * test can be run in parallel.
  *
  * Note however, that each thread will get its own copy of the DFDLTestSuite
+ *
+ * source is either a scala.xml.Elem, generally used for unit testing or a
+ * String. If the string starts with a / is it treated is if it were resoucrce
+ * on the classpath. Otherwise it is treated as if it were a URI.
  */
-class Runner private (elem: scala.xml.Elem, dir: String, file: String,
+class Runner private (
+  source: Either[scala.xml.Elem, String],
```

Comment:
You said you got rid of support for using an Elem here. Do we still need this Either then?
Ah. I see you still allow the whole test suite to be an Elem, just not a schema.

Reply:
I would not further change this. It is very convenient and a big advantage of Scala that one can just create Nodes for testing directly in a unit test. Much more convenient than dealing with separate file processing.